### PR TITLE
Update `gpg publish` to work with mirror arguments

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This PR updates the `gpg publish` command to work with the mirror
arguments, when trying to push keys to a mirror.

### CODE REVIEW

**Comment by douglasjacobsen** [comment]
@opadron Can you review this for me? You had a comment in the code about doing this, so it'd be good to see if this is how you think it should be done.

**Comment by tgamblin** [comment]
@opadron can you review this?

**Comment by douglasjacobsen** [comment]
Gentle poke on this. Let me know if you have any comments so I can make required modifications.

**Review by opadron — DISMISSED** [review]
This should be fine for now, but once #17547 lands (if I ever get around to finishing it!).  We'll need to update this handling code.

**Review by tgamblin — DISMISSED** [review]

**Comment by tgamblin** [comment]
talked to @douglasjacobsen -- he's going to add a test and then I think this is great!

**Comment by douglasjacobsen** [comment]
I added 3 tests (one for each mirror argument). So, I think this should be good to go, but let me know if you want any more tests to be added.

(This should also now kind of test `--rebuild-index` which wasn't covered before, but I'm not actually validating that the index is correct, just that it generates an index)

**Review by tgamblin — APPROVED** [review]

**Pull request merged**
